rust/ql/lib/codeql/rust/internal/Type.qll

```diff
@@ -15,6 +16,7 @@ newtype TType =
   TArrayType() or // todo: add size?
   TRefType() or // todo: add mut?
   TTypeParamTypeParameter(TypeParam t) or
+  TAssociatedTypeTypeParameter(TypeAlias t) { any(TraitItemNode trait).getADescendant() = t } or
```

Comment:
I have a slight preference for `getAnAssocItem` over `getADescendant`, but it should do that same.